redhat-developer/observability-operator
# Allow Operator to deploy and read resources

**Jira:** [MGDSTRM-10347](https://issues.redhat.com/browse/MGDSTRM-10347)

This change allows the Operator deploy and read the necessary configuration resources such as Prometheus rules and alerts. Currently the Operator can read these configurations from a GitHub repository only.

Currently the observability configuration secret can direct the Operator to a GitHub repo containing the required resources:
```
  access_token: <token-with-read-access-to-repo>
  channel: resources
  tag: <repo-tag>
  repository: >-
    <github-repo-url>
```

To direct the Operator to use a deployed image for configuration resources, the observability configuration secret needs to be configured:
```
  access_token: ''
  channel: resources
  tag: <image tag>
  repository: >-
    <image-registry-repo>
```

A pull secret may also be required to pull the configuration resources image from a private repo.

### Verification (for RHOSAK service)
* Apply the following values for observability configuration secret:
```
  access_token: ''
  channel: resources
  tag: v1.46.0
  repository: >-
    quay.io/rhoas/observability-resources-mk
```
* Apply relevant image pull secret, For RHOSAK, this secret should allow the Operator to pull from a private Quay repository in RHOAS org.
* Install the Operator through OLM. Image of this change available through `quay.io/vmanley/observability-operator-index:v4.0.5`
* Observability stack should install successfully with `obs-resources` deployment providing configuration resources.
* The Operator should have created a ConfigMap containing 6 rule files named `prometheus-obs-prometheus-rulefiles-0`. These should match the rule files from [observability-resources-mk repo](https://github.com/bf2fc6cc711aee1a0c2a/observability-resources-mk/tree/main/resources/prometheus)
* Deleting the observability stack and uninstalling the Operator should remove all resources including the `obs-resources` deployment and service
